common/components/stores/ProjectSearchStore.js

```diff
@@ -45,29 +52,47 @@ class ProjectSearchStore extends ReduceStore<State> {
 
   reduce(state: State, action: ProjectSearchActionType): State {
     switch (action.type) {
-      case 'FILTER_BY_KEYWORD':
-        fetch(new Request(this._getAPIURL(action.keyword)))
-          .then(response => response.json())
-          .then(projects =>
-            ProjectSearchDispatcher.dispatch({
-              type: 'SET_PROJECTS',
-              projects: List(projects.map(this._projectFromAPIData)),
-            }),
-          );
+      case 'INIT':
+        this._loadProjects({});
+        return state.set('projects', null);
+      case 'SET_ISSUE_AREA':
+        this._loadProjects({issueArea: action.issueArea});
+        return state.set('projects', null);
+      case 'SET_KEYWORD':
+        this._loadProjects({keyword: action.keyword});
         return state.set('projects', null);
-      case 'SET_PROJECTS':
+      case 'SET_PROJECTS_DO_NOT_CALL_OUTSIDE_OF_STORE':
         return state.set('projects', action.projects);
       default:
         (action: empty);
         return state;
     }
   }
 
-  _getAPIURL(keyword: ?string): string {
-    const baseURL = '/api/projects';
-    return keyword
-      ? baseURL + '?keyword=' + keyword
-      : baseURL;
+  _loadProjects(
+    {issueArea, keyword}: {issueArea?: ?IssueAreaType, keyword?: ?string}
+  ): void {
+    const url = [
+      '/api/projects?',
+      keyword ? '&keyword=' + keyword : null,
+      issueArea
+        ? '&issueArea=' + this._reformatIssueAreaForAPI(issueArea)
+        : null,
+    ].join('');
+    fetch(new Request(url))
+      .then(response => response.json())
+      .then(projects =>
+        ProjectSearchDispatcher.dispatch({
+          type: 'SET_PROJECTS_DO_NOT_CALL_OUTSIDE_OF_STORE',
+          projects: List(projects.map(this._projectFromAPIData)),
+        }),
+      );
+  }
+
+  _reformatIssueAreaForAPI(issueArea: IssueAreaType): string {
+    return issueArea
+      .replace(/\.?([A-Z])/g, (_, match) => "_" + match.toLowerCase())
```

Comment:
If we must use regexes, please comment on what they do.

Reply:
fair enough -- handled in 3c0daf9

Reply:
Check out _.snakeCase()
https://lodash.com/docs/4.17.4#snakeCase